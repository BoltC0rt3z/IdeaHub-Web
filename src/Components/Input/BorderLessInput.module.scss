.input {
  border: none;

  &:global(.ant-input:focus) {
    border: none;
    box-shadow: none;
  }
}
