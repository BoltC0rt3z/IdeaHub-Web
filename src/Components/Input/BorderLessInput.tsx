import React from 'react';
import { Input } from 'antd';
import { InputProps } from 'antd/lib/input';

import styles from './BorderLessInput.module.scss';

const BorderLessInput: React.FC<InputProps> = (props) => {
  const { className, ...restProps } = props;
  const classNames: string = [className, styles.input].join(' ');

  return <Input className={classNames} {...restProps} />;
};

export default BorderLessInput;
