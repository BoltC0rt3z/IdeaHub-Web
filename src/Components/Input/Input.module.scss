@import '../../styles/styles.module.scss';

@mixin input-border-colors {
  &:hover,
  &:active,
  &:focus {
    box-shadow: none;
    border-bottom: 1px solid $color-dark-blue-lighter;
    border-top: none;
    border-left: none;
    border-right: none;
  }
}

.Input {
  &:global(.ant-input) {
    border-top: none;
    border-left: none;
    border-right: none;
    background: $color-light;
    height: 56px;

    @include input-border-colors();
  }

  :global(.ant-input) {
    border-top: none;
    border-left: none;
    border-right: none;
    background: $color-light;
    height: 56px;

    @include input-border-colors();
  }

  &:global(.ant-input-affix-wrapper .ant-input:not(.ant-input-disabled)) {
    @include input-border-colors();
  }
}
