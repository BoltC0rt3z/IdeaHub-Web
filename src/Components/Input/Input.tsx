import React from "react";
import { Input as AntdInput } from "antd";
import { InputProps } from "antd/lib/input";

import styles from "./Input.module.scss";

const Input: React.FC<InputProps> = props => {
  return <AntdInput {...props} className={styles.input} autoComplete="off" />;
};

export default Input;
