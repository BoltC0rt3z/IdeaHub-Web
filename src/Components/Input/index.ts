import Input, { PasswordInput } from './Input';
import BorderLessInput from './BorderLessInput';

export { Input, PasswordInput, BorderLessInput };
