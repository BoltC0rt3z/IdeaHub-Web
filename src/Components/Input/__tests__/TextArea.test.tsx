import React from 'react';
import { ReactWrapper, mount } from 'enzyme';

import { TextArea } from '../Input';

describe('Tests for text area', () => {
  const wrapper: ReactWrapper = mount(<TextArea />);

  it('Should render text area without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
