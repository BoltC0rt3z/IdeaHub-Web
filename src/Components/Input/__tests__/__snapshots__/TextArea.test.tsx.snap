// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for text area Should render text area without crushing 1`] = `ReactWrapper {}`;
